apps/keyboard/js/imes/emoji/emoji.js

```diff
@@ -0,0 +1,90 @@
+/* global InputMethods, PromiseStorage*/
+
+
+'use strict';
+
+(function() {
+  function RecentKeys() {
+    this._items = [];
+  }
+
+  RecentKeys.prototype.RECENT_KEY_COUNT = 18;
+
+  RecentKeys.prototype.getItems = function() {
+    return this._items || [];
+  };
+
+  RecentKeys.prototype.setItems = function(val) {
+    if (Array.isArray(val) && val.length) {
+      this._items = val.slice(0, this.RECENT_KEY_COUNT);
+    }
+  };
+
+  RecentKeys.prototype.getPanelKeys = function() {
+    return this._items.map(function(val) {
```

Comment:
Then you would need to `throw` at `getItems` (instead of returning an empty array) :)
We don't want to fail silently -- it would cost us time to find bugs in the future...